pkg/cluster/base.go

```diff
@@ -1314,3 +1315,23 @@ func (p *ProviderBase) ValdiateRequireSSHPrivateKey() error {
 	}
 	return nil
 }
+
+func (p *ProviderBase) GenerateDisabledComponent() string {
+	if p.K3sVersion == "" {
+		return "--disable"
+	}
+	noDeployVersion, err := semver.NewVersion("v1.19.1+k3s1")
```

Comment:
Maybe this can be moved outside of the function because this only need to be parsed one time. And add comments for the reason why we are using `v1.19.1+k3s1` as the `noDeployVersion`.